resilience4j/src/main/java/com/linecorp/armeria/resilience4j/circuitbreaker/Resilience4JCircuitBreakerClientHandler.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.resilience4j.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerClient;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerClientHandler;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerRule;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerRuleWithContent;
+import com.linecorp.armeria.client.circuitbreaker.ClientCircuitBreakerGenerator;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.circuitbreaker.CircuitBreakerCallback;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+
+/**
+ * A {@link CircuitBreakerClientHandler} implementation for use with Resilience4j's {@link CircuitBreaker}.
+ */
+public final class Resilience4JCircuitBreakerClientHandler implements CircuitBreakerClientHandler<HttpRequest> {
+
+    private static final Logger logger =
+            LoggerFactory.getLogger(Resilience4JCircuitBreakerClientHandler.class);
+
+    /**
+     * Creates a new decorator using the specified {@link CircuitBreaker} instance and
+     * {@link CircuitBreakerRule}.
+     *
+     * <p>Since {@link CircuitBreaker} is a unit of failure detection, don't reuse the same instance for
+     * unrelated services.
+     */
+    public static Function<? super HttpClient, CircuitBreakerClient>
+    newDecorator(CircuitBreaker circuitBreaker, CircuitBreakerRule rule) {
+        requireNonNull(circuitBreaker, "circuitBreaker");
+        return newDecorator((ctx, req) -> circuitBreaker, rule);
+    }
```

Comment:
Oops 😅. I don't think this was an intentional change.
Let me update/clean up this PR after #4426 is merged since it's difficult to view diffs at the moment 🙏